Dockerfile

```diff
@@ -1,30 +1,53 @@
 FROM python:3.6-slim
-MAINTAINER Product Delivery irc://irc.mozilla.org/#product-delivery
+MAINTAINER Product Delivery irc://irc.mozilla.org/#storage-team
 
-RUN \
-    apt-get update && \
-    apt-get -y install zip make gcc && \
-    rm -rf /var/lib/apt/lists/* && \
-    pip install virtualenv
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONPATH=/app/
+
+# add a non-privileged user for installing and running the application
+# don't use --create-home option to prevent populating with skeleton files
+RUN mkdir /app && \
+    chown 10001:10001 /app && \
+    groupadd --gid 10001 app && \
+    useradd --no-create-home --uid 10001 --gid 10001 --home-dir /app app
+
+# install a few essentials and clean apt caches afterwards
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        apt-transport-https build-essential curl git zip
+
+# Clean up apt
+RUN apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 COPY . /app
+
+# Install Python dependencies
+RUN pip install -e app/jobs/
```

Comment:
As far as I understood your approach, this does not install Python dependencies at all (move comment below?)

Reply:
Correct. It just installs `buildhub` itself.